spec/models/post_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  subject { Post.new(author_id: 1, title: 'My first post', body: 'This is my first post') }
+
+  before { subject.save }
+
+  it 'Validates the presence of Title' do
+    subject.title = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'Validates comment counter to not be a string' do
+    subject.comment_count = 'great'
+    expect(subject).to_not be_valid
+  end
+
+  it 'Validates the length of Title to be less than or equal to 250' do
+    subject.title = 'a' * 251
+    expect(subject).to_not be_valid
+  end
+
+  it 'Validates increament of post counter after creation of post' do
+    user = User.create(name: 'Philip', photo: 'https://...', bio: 'physician from Britain')
+    expect(user.post_count).to eq(1)
+  end
```

Comment:
- [ ] [Optional] The `update_post_count` method for which this unit test was written returns the wrong result. It should update the `posts_count` attribute in the `User` model. 

```ruby on rails
 def update_post_count
    user.increment!(:posts_count)
  end
```